assets/js/controllers/settings/addresses.controller.js

```diff
@@ -2,75 +2,86 @@ angular
   .module('walletApp')
   .controller('SettingsAddressesCtrl', SettingsAddressesCtrl);
 
-function SettingsAddressesCtrl ($scope, $state, $stateParams, $q, $sce, Wallet, MyWalletHelpers, MyBlockchainApi, Alerts, paymentRequests, $uibModal) {
-  $scope.paymentRequests = paymentRequests;
+function SettingsAddressesCtrl ($scope, $translate, $rootScope, $state, $stateParams, $q, $sce, Wallet, Labels, MyWalletHelpers, MyBlockchainApi, Alerts, $uibModal) {
+  if (!Wallet.status.isLoggedIn) {
+    console.error('Controller depends on being logged in');
+    return;
+  }
+
+  let accountIndex = parseInt($stateParams.account, 10);
+
+  $scope.presentFilter = (address) => address && !!address.label && address.used === false;
+
+  $scope.pastAddressesPage = (page, pageLength) => {
+    let pastFilter = (address) => address && (!address.label || address.used === true);
+
+    return $scope.addresses.filter(pastFilter)
+                           .reverse().slice((page - 1) * pageLength, page * pageLength);
+  };
```

Comment:
I also just noticed the duplication issue. Will fix that. Hopefully delete is the same issue...

I'll look into the issue of the first available address in the past addresses section.